### ISSUE TITLE
Bugged Submarine Hatch door.

### ISSUE THREAD

**Issue by MesssaTee** [Regalis11/Barotrauma#10851]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I was attacked by some enemies a little before this, and I'm guessing the hatch door was open when this happened, but the airlock door to the sub was destroyed, and I'm guessing this caused the lock switch to get messed up when the airlock door was repaired, since now when I try to, I cannot leave the submarine through the hatch door even though it's opening and water is rushing through the hatch. I'm in single-player, so I know ping is not the issue. 

EDIT: The said issue is resolved by you destroying the hatch door and repairing it (most likely with the airlock door closed), that should cause the collision to fix itself. This leads me to believe that the hatch door was opened whilst the airlock door was being fixed, because the entire sub was flooded, there was practically no point in me closing the hatch door as it would only waste time.

### Reproduction steps

(only a theory, still unsure how this happened)
1. have the hatch door to the outside of the sub opened
2. have the airlock door get destroyed
3. repair the airlock door with the hatch door opened/closed
4. should be enough to activate the bug

### Bug prevalence

Just once

### Version

0.20.16.1

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I unfortunately couldn't find a way to reproduce the issue. Tested in both single player and multiplayer. I couldn't come up with a way to damage an open door, other than switching to the sub editor mid-round and setting the condition to 0. When the door breaks, it seems to force it shut, and it remains shut when it's repaired.

Did the bug happen in single player or multiplayer? Do you recall which submarine you were using (one of the vanilla subs or a custom one)?

**Comment by Regalis11** [comment]
No known steps to repro, closing

**Comment by PquePC** [comment]
Hey! I just had the same bug on a multiplayer session, however, I'm not sure if MesssaTee was right about the reason. I was playing with some friends on a Dugong and we docked to a station. When we tried to enter the station the hatch was "flickering" open and closed. We're using a full vanilla Dugong, so we didn't change any of the logic. I disconnected the toggle_state wire and the hatch stayed open. Then we connected it again when we undocked to prevent flooding, and there was no issue, until we docked again.

Basically, I'm not sure if I would call this a bug, as I think it might be due to an issue with the logic behind the docking hatch, as the issue only happens (at least to us) when we're docked. I can give it a deeper look myself (I have a pretty good understanding of logic), but I can't seem to find where the hatch's logic is hidden. Maybe I should try on the editor.
